File: Backend/NootsWEB/Noots.Editor/Entities/Text/TextDiff.cs

using Common.DatabaseModels.Models.NoteContent.TextContent;

namespace Noots.Editor.Entities.Text;

public class TextDiff
{
    public Guid ContentId { set; get; }

    public HTypeENUM? HeadingTypeId { set; get; }

    public NoteTextTypeENUM? NoteTextTypeId { set; get; }

    public bool? Checked { set; get; }

    public List<BlockDiff>? BlockDiffs { set; get; }
}

File: Client/src/app/content/notes/full-note/models/api/editor/text-diff.ts
import { BaseText } from "src/app/content/notes/models/editor-models/base-text";
import { ContentTypeENUM } from "src/app/content/notes/models/editor-models/content-types.enum";
import { HeadingTypeENUM } from "src/app/content/notes/models/editor-models/text-models/heading-type.enum";
import { NoteTextTypeENUM } from "src/app/content/notes/models/editor-models/text-models/note-text-type.enum";
import { TextBlock } from "src/app/content/notes/models/editor-models/text-models/text-block";

export class TextDiff {
    id: string;

    typeId: ContentTypeENUM;

    order: number;

    headingTypeId?: HeadingTypeENUM;

    noteTextTypeId: NoteTextTypeENUM;

    checked?: boolean;

    contents: TextBlock[];

    initFrom(text: BaseText) : TextDiff {
        this.id = text.id;
        this.typeId =  text.typeId;
        this.order = text.order;
        this.headingTypeId = text.headingTypeId;
        this.noteTextTypeId = text.noteTextTypeId;
        this.checked = text.checked;
        this.contents = text.contents;
        return this;
    }
}